assets/activation-scripts/activate.d/flox-env-helper.awk

```diff
@@ -0,0 +1,194 @@
+  # - If MANPATH does not start or end with `:`, -> rule 4,
+  #   FLOX_ENV:MANPATH replaces the defaults (no change)
+  new_manpath = remove_path_dups(_prepend_manpath ENVIRON["MANPATH"] ":")
+
+  # Emit commands to set variables for the various shell dialects.
+  if (_flox_shell == "bash") {
+    print "export FLOX_ENV_DIRS=\"" new_flox_env_dirs "\";"
+    print "export FLOX_ENV_LIB_DIRS=\"" new_flox_env_lib_dirs "\";"
+    print "export PATH=\"" new_path "\";"
+    print "export MANPATH=\"" new_manpath "\";"
+  } else if (_flox_shell == "tcsh") {
+    print "setenv FLOX_ENV_DIRS \"" new_flox_env_dirs "\";"
+    print "setenv FLOX_ENV_LIB_DIRS \"" new_flox_env_lib_dirs "\";"
+    print "setenv PATH \"" new_path "\";"
+    print "setenv MANPATH \"" new_manpath "\";"
+  } else if (_flox_shell == "fish") {
+    print "set -gx FLOX_ENV_DIRS \"" new_flox_env_dirs "\";"
+    print "set -gx FLOX_ENV_LIB_DIRS \"" new_flox_env_lib_dirs "\";"
+    print "set -gx PATH \"" new_path "\";"
+    print "set -gx MANPATH \"" new_manpath "\";"
+  } else if (_flox_shell == "zsh") {
+    print "export FLOX_ENV_DIRS=\"" new_flox_env_dirs "\";"
+    print "export FLOX_ENV_LIB_DIRS=\"" new_flox_env_lib_dirs "\";"
+    print "export PATH=\"" new_path "\";"
+    print "export MANPATH=\"" new_manpath "\";"
+  } else {
+    print("Unknown shell: " flox_shell) > "/dev/stderr"
+    exit 1
+  }
+
+  tracer("END")
+}
+
+# No input block means script will exit immediately after BEGIN block.
+
+# Function definitions follow.
+#
+# A note regarding function arguments, per the awk man page:
+#   Parameters are local to the function; all other variables are global.
+#   Thus local variables may be created by providing excess parameters in
+#   the function definition.
+#
+# So the presence of "extra" parameters in the following function
+# definitions is intentional and for declaring local variables.
+
+function append_dirs(string, path, path_array, i, result)
+{
+  # local variables: path_array, i, result
+  result = ""
+  for (i = 1; i <= split(path, path_array, ":"); i++) {
+    if (result == "") {
+      result = path_array[i] string
+    } else {
+      result = result ":" path_array[i] string
+    }
+  }
+  return result
+}
+
+function basename(file, a, n)
+{
+  # local variables: a, n
+  n = split(file, a, "/")
+  return a[n]
+}
+
+# Inserts provided value at beginning of path provided it is
+# not already found in the path. This does not remove duplicates.
+function prepend_if_not_found(insert, path, path_array, seen_hash, i, result)
+{
+  # local variables: path_array, seen_hash, i, result
+  for (i = 1; i <= split(path, path_array, ":"); i++) {
+    seen_hash[path_array[i]] = 1
+  }
+  if (seen_hash[insert] == 1) {
+    result = path
+  } else if (path == "") {
+    result = insert
+  } else {
+    result = insert ":" path
+  }
+  return result
+}
+
+function remove_path_dups(path, path_array, dedup_array, seen_hash, i, dedup_count, result)
```

Comment:
Nit: I'd still prefer marking local variables somehow e.g. `_var` just an inline comment is easily outdated or overlooked.

Reply:
Ah, OK. I agree for all but `i` as that's just a universal counter. Will update accordingly.  👍

Reply:
Renamed local variables in commit 7da7cc86ebad225674de34a3bbe075b70ad6fb77